aws/data_source_aws_eks_cluster_auth_test.go

```diff
@@ -0,0 +1,66 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSEksClusterAuthDataSource_basic(t *testing.T) {
+	dataSourceResourceName := "data.aws_eks_cluster_auth.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAwsEksClusterAuthConfig_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsEksClusterAuthToken(dataSourceResourceName),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsEksClusterAuthToken(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Can't find EKS Cluster Auth resource: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("EKS Cluster Auth resource ID not set")
+		}
+
+		name := rs.Primary.Attributes["name"]
+		if expected := "foobar"; name != expected {
```

Comment:
Checking the Terraform state for a specific attribute value can be performed without custom logic in the `TestStep` via:

```go
resource.TestCheckResourceAttr(dataSourceResourceName, "name", "foobar"),
```